packages/sitecore-jss-react/src/components/Link.tsx

```diff
@@ -38,88 +38,92 @@ export type LinkProps = React.DetailedHTMLProps<
 
-    // in an ideal world, we'd pre-render React children here and inject them between editableFirstPart and editableLastPart.
-    // However, we cannot combine arbitrary unparsed HTML (innerHTML) based components with actual vDOM components (the children)
-    // because the innerHTML is not parsed - it'd make a discontinuous vDOM. So, we'll go for the next best compromise of rendering the link field and children separately
-    // as siblings. Should be "good enough" for most cases - and write your own helper if it isn't. Or bring xEditor out of 2006.
+    if (!link) {
+      return null;
+    }
 
-    const htmlProps = {
-      className: 'sc-link-wrapper',
-      dangerouslySetInnerHTML: {
-        __html: markup,
-      },
-      ...otherProps,
-      key: 'editable',
+    const anchorAttrs: { [attr: string]: unknown } = {
+      href: link.querystring ? `${link.href}?${link.querystring}` : link.href,
+      className: link.class,
+      title: link.title,
+      target: link.target,
     };
 
-    resultTags.push(<span {...htmlProps} />);
-
-    // don't render normal link tag when editing, if no children exist
-    // this preserves normal-ish behavior if not using a link body (no hacks required)
-    if (!children) {
-      return resultTags[0];
+    if (anchorAttrs.target === '_blank' && !anchorAttrs.rel) {
+      // information disclosure attack prevention keeps target blank site from getting ref to window.opener
+      anchorAttrs.rel = 'noopener noreferrer';
     }
-  }
-
-  // handle link directly on field for forgetful devs
-  const link = (dynamicField as LinkFieldValue).href
-    ? (field as LinkFieldValue)
-    : (dynamicField as LinkField).value;
 
-  if (!link) {
-    return null;
-  }
+    const linkText = showLinkTextWithChildrenPresent || !children ? link.text || link.href : null;
 
-  const anchorAttrs: { [attr: string]: unknown } = {
-    href: link.querystring ? `${link.href}?${link.querystring}` : link.href,
-    className: link.class,
-    title: link.title,
-    target: link.target,
-  };
+    resultTags.push(
+      React.createElement(
+        'a',
+        { ...anchorAttrs, ...otherProps, key: 'link', ref },
+        linkText,
+        children
+      )
+    );
 
-  if (anchorAttrs.target === '_blank' && !anchorAttrs.rel) {
-    // information disclosure attack prevention keeps target blank site from getting ref to window.opener
-    anchorAttrs.rel = 'noopener noreferrer';
+    return <React.Fragment>{resultTags}</React.Fragment>;
   }
-
-  const linkText = showLinkTextWithChildrenPresent || !children ? link.text || link.href : null;
-
-  resultTags.push(
-    React.createElement('a', { ...anchorAttrs, ...otherProps, key: 'link' }, linkText, children)
-  );
-
-  return <React.Fragment>{resultTags}</React.Fragment>;
-};
+);
 
 export const LinkPropTypes = {
   field: PropTypes.oneOfType([
     PropTypes.shape({
       href: PropTypes.string,
     }),
+    PropTypes.shape({
```

Comment:
Needed to make this optional due to React's forwardRef parsing this into a WeakValidationMap which makes the field's url null. Happy to change this, but no idea on how to make this use a stricter type. :(